test/unittests/capi_test.cpp

```diff
@@ -1195,6 +1195,127 @@ TEST(capi, get_type)
+    ASSERT_NE(module, nullptr);
+
+    EXPECT_EQ(fizzy_get_import_count(module), 4);
+    fizzy_free_module(module);
+}
+
+TEST(capi, get_import_description)
+{
+    /* wat2wasm
+      (func (import "m" "f1") (result i32))
+      (func (import "m" "f2") (param i64))
+      (global (import "m" "g1") i32)
+      (global (import "m" "g2") (mut f64))
+      (table (import "m" "t") 10 anyfunc)
+      (memory (import "m" "mem") 1 4)
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000109026000017f60017e00023106016d0266310000016d0266320001016d026731037f0001"
+        "6d026732037c01016d01740170000a016d036d656d02010104");
+
+    const auto* module = fizzy_parse(wasm.data(), wasm.size());
+    ASSERT_NE(module, nullptr);
+    ASSERT_EQ(fizzy_get_import_count(module), 6);
+
+    const auto import0 = fizzy_get_import_description(module, 0);
+    EXPECT_STREQ(import0.module, "m");
+    EXPECT_STREQ(import0.name, "f1");
+    EXPECT_EQ(import0.kind, FizzyExternalKindFunction);
+    EXPECT_EQ(import0.desc.function_type.inputs_size, 0);
+    EXPECT_EQ(import0.desc.function_type.output, FizzyValueTypeI32);
+
+    const auto import1 = fizzy_get_import_description(module, 1);
+    EXPECT_STREQ(import1.module, "m");
+    EXPECT_STREQ(import1.name, "f2");
+    EXPECT_EQ(import1.kind, FizzyExternalKindFunction);
+    ASSERT_EQ(import1.desc.function_type.inputs_size, 1);
+    EXPECT_EQ(import1.desc.function_type.inputs[0], FizzyValueTypeI64);
+    EXPECT_EQ(import1.desc.function_type.output, FizzyValueTypeVoid);
+
+    const auto import2 = fizzy_get_import_description(module, 2);
+    EXPECT_STREQ(import2.module, "m");
+    EXPECT_STREQ(import2.name, "g1");
+    EXPECT_EQ(import2.kind, FizzyExternalKindGlobal);
+    EXPECT_EQ(import2.desc.global_type.value_type, FizzyValueTypeI32);
+    EXPECT_FALSE(import2.desc.global_type.is_mutable);
+
+    const auto import3 = fizzy_get_import_description(module, 3);
+    EXPECT_STREQ(import3.module, "m");
+    EXPECT_STREQ(import3.name, "g2");
+    EXPECT_EQ(import3.kind, FizzyExternalKindGlobal);
+    EXPECT_EQ(import3.desc.global_type.value_type, FizzyValueTypeF64);
+    EXPECT_TRUE(import3.desc.global_type.is_mutable);
+
+    const auto import4 = fizzy_get_import_description(module, 4);
+    EXPECT_STREQ(import4.module, "m");
+    EXPECT_STREQ(import4.name, "t");
+    EXPECT_EQ(import4.kind, FizzyExternalKindTable);
+    EXPECT_EQ(import4.desc.table_limits.min, 10);
+    EXPECT_FALSE(import4.desc.table_limits.has_max);
+
+    const auto import5 = fizzy_get_import_description(module, 5);
+    EXPECT_STREQ(import5.module, "m");
+    EXPECT_STREQ(import5.name, "mem");
+    EXPECT_EQ(import5.kind, FizzyExternalKindMemory);
+    EXPECT_EQ(import5.desc.memory_limits.min, 1);
+    EXPECT_TRUE(import5.desc.memory_limits.has_max);
+    EXPECT_EQ(import5.desc.memory_limits.max, 4);
+
+    fizzy_free_module(module);
+}
+
+TEST(capi, import_name_after_instantiate)
```

Comment:
Is this testing the lifetime of the name/namespace pointers?

Reply:
Yes, checks that it's alive after instantiation, as the documentation claims.